packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -325,6 +326,30 @@ export const defaultEmbedding = new LangChainEmbeddingWrapper(
   new OpenAIEmbeddings({ openAIApiKey: process.env.OPENAI_API_KEY })
 );
 
+/** A function converting documents into EmbeddedChunks. Also see `asVectorizer`. */
+export type Vectorizer<
+  DocumentMetadata extends Jsonifiable = Jsonifiable,
+  ChunkMetadata extends Jsonifiable = Jsonifiable
+> = (documents: Document<DocumentMetadata>[]) => Promise<EmbeddedChunk<ChunkMetadata>[]>;
+
+/** Converts a Chunker + Embedding into a unified Vectorizer function. */
+export function asVectorizer<
+  DocumentMetadata extends Jsonifiable = Jsonifiable,
+  ChunkMetadata extends Jsonifiable = Jsonifiable
+>(chunker: Chunker<DocumentMetadata, ChunkMetadata>, embedding: Embedding) {
+  return async (docs: Document<DocumentMetadata>[]) => {
+    const chunks: Chunk<ChunkMetadata>[] = [];
+    await Promise.all(
+      docs.map(async (doc) => {
+        const docChunks = await chunker(doc);
+        chunks.push(...docChunks);
+      })
+    );
```

Comment:
```suggestion
    const chunks = _.flat(await Promise.all(
      docs.map(async (doc) => chunker(doc))
    ));
```

I think this will work.

Reply:
What is the underscore here? Is that something I need to import?  (This didn't seem to work without any new imports fwiw.)

Reply:
Yep! It's not a special symbol although I see how it could look that way. :smile:

```tsx
import _ from 'lodash`
```